lib/logflare/backends.ex

```diff
@@ -93,12 +92,12 @@ defmodule Logflare.Backends do
   defp typecast_config_string_map_to_atom_map(nil), do: nil
 
   defp typecast_config_string_map_to_atom_map(%SourceBackend{type: type} = source_backend) do
```

Comment:
an aside: we can consider using [Params](https://hexdocs.pm/params/readme.html) in the future for config typecasting, which would be safer and cleaner than what we are doing now.

Reply:
yes, it will also simplify a lot validation since we can just run a validate required